docs: Updated Fluent Bit docs to use official plugin + Sandbox (#14004)
grafana/loki: docs/sources/send-data/fluentbit/community-plugin.md

---
title: Fluent Bit Community Plugin
menuTitle:  Fluent Bit Community Plugin
description: Provides instructions for how to install, configure, and use the Fluent Bit Community plugin to send logs to Loki.
aliases: 
- ../clients/fluentbit/
weight:  500
---
# Fluent Bit Community Plugin

{{< admonition type="warning" >}}

> We recommend using the official [Fluent Bit Loki plugin]({{< relref "./fluent-bit-plugin.md" >}}). The official plugin is more feature-rich and has better support for features such as structured metadata. The community plugin is still available for use, but it's no longer actively maintained.

Reviewer: ```suggestion
We recommend using the official [Fluent Bit Loki plugin](https://grafana.com/docs/loki/<LOKI_VERSION>/send-data/fluentbit/fluent-bit-plugin/). The official plugin is more feature-rich and has better support for features such as structured metadata. The community plugin is still available for use, but it's no longer actively maintained.
```

{{< /admonition >}}

> The Fluent Bit Community plugin by Grafana Labs (`grafana-loki`) provided an alternative way to send logs to Loki. Although very similar to the [official plugin]({{< relref "./fluent-bit-plugin.md" >}}) there are some differences in the configuration options. This page provides instructions for how to install, configure, and use the Fluent Bit Community plugin to send logs to Loki. Although the plugin is no longer actively maintained, this documentation is still available for reference.

Reviewer: ```suggestion
The Fluent Bit Community plugin by Grafana Labs (`grafana-loki`) provided an alternative way to send logs to Loki. Although very similar to the [official plugin](https://grafana.com/docs/loki/<LOKI_VERSION>/send-data/fluentbit/fluent-bit-plugin/) there are some differences in the configuration options. This page provides instructions for how to install, configure, and use the Fluent Bit Community plugin to send logs to Loki. Although the plugin is no longer actively maintained, this documentation is still available for reference.
```

{{< youtube id="s43IBSVyTpQ" >}}

## Usage

### Docker

You can run a Fluent Bit container with Loki output plugin pre-installed using our [Docker Hub](https://hub.docker.com/r/grafana/fluent-bit-plugin-loki) image:

```bash
docker run -v /var/log:/var/log \
    -e LOG_PATH="/var/log/*.log" -e LOKI_URL="http://localhost:3100/loki/api/v1/push" \
    grafana/fluent-bit-plugin-loki:latest
```

Or, an alternative is to run the fluent-bit container using [Docker Hub](https://hub.docker.com/r/fluent/fluent-bit) image:

> ### Docker Container Logs

Reviewer: ```suggestion
### Docker container logs
```

        multiline.parser docker, cri
        Mem_Buf_Limit 5MB
        Skip_Long_Lines On

  outputs: |
    [Output]
        Name grafana-loki
        Match kube.*
        Url ${FLUENT_LOKI_URL}
        Labels {job="fluent-bit"}
        LabelKeys level,app # this sets the values for actual Loki streams and the other labels are converted to structured_metadata https://grafana.com/docs/loki/<LOKI_VERSION>/get-started/labels/structured-metadata/
        BatchWait 1
        BatchSize 1001024
        LineFormat json
        LogLevel info
        AutoKubernetesLabels true
```

```bash
helm repo add fluent https://fluent.github.io/helm-charts
helm repo update
helm install fluent-bit fluent/fluent-bit -f values.yaml
```

By default it will collect all containers logs and extract labels from Kubernetes API (`container_name`, `namespace`, etc.).

If you also want to host your Loki instance inside the cluster install the [official Loki Helm chart](https://grafana.com/docs/loki/<LOKI_VERSION>/setup/install/helm/).

### AWS Elastic Container Service (ECS)

You can use the fluent-bit Loki Docker image as a Firelens log router in AWS ECS.

> For more information about this see our [AWS documentation]({{< relref "../promtail/cloud/ecs" >}})

Reviewer: ```suggestion
For more information about this see our [AWS documentation](https://grafana.com/docs/loki/<LOKI_VERSION>/send-data/promtail/cloud/ecs/).
```

### Local

First, you need to follow the [instructions](https://github.com/grafana/loki/blob/main/clients/cmd/fluent-bit/README.md) in order to build the plugin dynamic library.

Assuming you have Fluent Bit installed in your `$PATH` you can run the plugin using:

```bash
fluent-bit -e /path/to/built/out_grafana_loki.so -c fluent-bit.conf
```

You can also adapt your plugins.conf, removing the need to change the command line options:

```conf
[PLUGINS]
    Path /path/to/built/out_grafana_loki.so
```

> ## Configuration Options

Reviewer: ```suggestion
## Configuration options
```

| DqueName             | Queue name, must be uniq per output                                                                                                                                                                                                                                                                                                                                                     | dque                                   |

### Labels

Labels are used to [query logs]({{< relref "../../query" >}}) `{container_name="nginx", cluster="us-west1"}`, they are usually metadata about the workload producing the log stream (`instance`, `container_name`, `region`, `cluster`, `level`).  In Loki labels are indexed consequently you should be cautious when choosing them (high cardinality label values can have performance drastic impact).

> You can use `Labels`, `RemoveKeys` , `LabelKeys` and `LabelMapPath` to how the output plugin will perform labels extraction.

Reviewer: Missing a word here?  Sorry, I missed seeing this again before I approved.

---
title: Fluent Bit Community Plugin
menuTitle:  Fluent Bit Community Plugin
description: Provides instructions for how to install, configure, and use the Fluent Bit Community plugin to send logs to Loki.
aliases: 
- ../clients/fluentbit/
weight:  500
---
# Fluent Bit Community Plugin

{{< admonition type="note" >}}

> It is now recommended to use the official [Fluent Bit Loki plugin]({{< relref "./_index.md" >}}). The official plugin is more feature-rich and has better support for features such as structured meta data. The community plugin is still available for use, but it is no longer actively maintained.

Reviewer: ```suggestion
We recommend using the official [Fluent Bit Loki plugin]({{< relref "./_index.md" >}}). The official plugin is more feature-rich and has better support for features such as structured meta data. The community plugin is still available for use, but it is no longer actively maintained.
```
